src/cmd/cli/command/compose.go

```diff
@@ -193,14 +193,14 @@ func makeComposeUpCmd() *cobra.Command {
 				}
 
 				var errDeploymentFailed pkg.ErrDeploymentFailed
-				if errors.As(context.Cause(tailCtx), &errDeploymentFailed) {
+				if errors.As(context.Cause(tailCtx), &errDeploymentFailed) || errors.As(err, &errDeploymentFailed) {
 					// Tail got canceled because of deployment failure: prompt to show the debugger
 					term.Warn(errDeploymentFailed)
 					if !nonInteractive {
 						failedServices := []string{errDeploymentFailed.Service}
 						track.Evt("Debug Prompted", P("failedServices", failedServices), P("etag", deploy.Etag), P("reason", errDeploymentFailed))
 						// Call the AI debug endpoint using the original command context (not the tailCtx which is canceled)
-						if nil == cli.InteractiveDebug(cmd.Context(), client, provider, deploy.Etag, project, failedServices) {
+						if nil == cli.InteractiveDebug(cmd.Context(), client, provider, deploy.Etag, project, failedServices, since) {
```

Comment:
Perhaps we should be passing a `cli.DebugConfig` struct here as well?
```suggestion

						err = cli.InteractiveDebug(cmd.Context(), cli.DebugConfig{
							Project:  project,
							Provider: provider,
							Etag:     etag,
							Client:   client,
						})
						if err == nil {
```